Add pull mode to remote job execution (#1630)
theforeman/foreman-documentation: guides/common/modules/proc_migrating-a-host-to-use-pull-mode.adoc

> [id="migrating-a-host-to-use-pull-mode_{context}"]

Reviewer: I think this is sufficiently covered by https://github.com/theforeman/foreman-documentation/pull/1548 so this file can be removed.

[id="migrating-a-host-to-use-pull-mode_{context}"]
= Migrating a Host to Use Pull Mode

To migrate a host from `katello-agent` to pull-based transport, you can use the `katello-pull-transport-migrate` utility.

.Prerequisites
* The host runs an Enterprise Linux derivative operating system.
ifndef::satellite[]
The utility should work on any system that uses Red Hat Subscription Manager, but it was tested only on Enterprise Linux.
endif::[]
* The host is registered to Katello.

> The utility will exit if the host is registered to Red Hat Subscription Manager directly.

Reviewer 2: ```suggestion
The utility will not work if the host is registered to Red Hat Subscription Manager directly.
```

I assume "exit" means abort/stop?

Author: This sentence is supposed to clarify why we mention Katello instead of {Project}. Does it help?

Reviewer 2: OK, then please reword this. I might create a separate PR to rename Katello in the future. But this is definitely good for now. 👍

* You have enabled the {Team} {project-client-name} repository on the host.

.Procedure
. Install the `katello-pull-transport-migrate` package on the host.
You can select one of the following options:
** Add the package to a Content View that is assigned to the target host, promote the Content View, and then install the package.
For more information, see {ContentManagementDocURL}Managing_Content_Views_content-management[Managing Content Views] and {ContentManagementDocURL}Installing_Packages_on_a_Host_content-management[Installing Packages on a Host] in _{ContentManagementDocTitle}_.
** Execute a remote job on the target host to install the package.
** Connect to the host using SSH and install the package manually:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
# yum install katello-pull-transport-migrate

> ----

Author: @maximiliankolb Do you think the example is necessary? I'm wondering if I remove it, it will be less confusing because of different host OS versions...

> ** Connect to the host using SSH and install the package manually:
> +
> [options="nowrap", subs="+quotes,verbatim,attributes"]
> ----
> # yum install katello-pull-transport-migrate
> ----

Reviewer 2: ```suggestion
** Connect to the host using SSH and install `katello-pull-transport-migrate` manually.
```
This would be an alternative to an OS-specific example.

[id="migrating-a-host-to-use-pull-mode_{context}"]
= Migrating a Host to Use Pull Mode

To migrate a host from `katello-agent` to pull-based transport, you can use the `katello-pull-transport-migrate` utility.

.Prerequisites
* The host runs an Enterprise Linux derivative operating system.
ifndef::satellite[]
The utility should work on any system that uses Red Hat Subscription Manager, but it was tested only on Enterprise Linux.
endif::[]

> * The host is registered to Katello.

Reviewer 2: ```suggestion
* The host is registered to {Project}.
```

Author: I'm not sure if it's a good idea to replace this, because the utility mentions in prerequisites specifically:
> Registered to Katello (script will exit if client is registered directly to RHSM)

Author: @adamruzicka Can you weigh in, please?

Reviewer 2: This is still open IMHO.

Author: I tried to clarify that with the additional sentence. Does it help?

Author: Oh, sorry, I forgot to mention...
I discussed it with @adamruzicka on IRC and we came to this conclusion: The situation that the host is not registered via Katello is possible, so saying Katello instead of {Project} is more precise and probably appropriate in this case.

* You have enabled the {Team} {project-client-name} repository on the host.

.Procedure
. Install the `katello-pull-transport-migrate` package on the host.
You can select one of the following options:
** Add the package to a Content View that is assigned to the target host, promote the Content View, and then install the package.
For more information, see {ContentManagementDocURL}Managing_Content_Views_content-management[Managing Content Views] and {ContentManagementDocURL}Installing_Packages_on_a_Host_content-management[Installing Packages on a Host] in _{ContentManagementDocTitle}_.
** Execute a remote job on the target host to install the package.
** Connect to the host using SSH and install the package manually:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
# yum install katello-pull-transport-migrate
----

> 

Reviewer 3: ```suggestion
```
Sorry, I believe that in my first review I accidentally suggested removing a different line. I meant to suggest removing this line.

Author: Actually, the empty line is necessary here for proper formatting.

> yum install katello-pull-transport-migrate

Reviewer 3: ```suggestion
# yum install katello-pull-transport-migrate
```

+
The package installs `foreman_ygg_worker` and `yggdrasil` as dependencies and enables the pull mode on the host.
. To verify the installation was successful, execute the following command on the host:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> systemctl status yggdrasild

Reviewer 3: ```suggestion
# systemctl status yggdrasild
```

----
+
The `yggdrasild` service must be running on the host in order to execute remote jobs in pull mode.
. After the package is installed, it is safe to remove `katello-agent` from the host.

> 

Reviewer 3: ```suggestion
```
